### DISCUSSION TITLE
Create Child SA after the  ipsec tunnel establishment

### DISCUSSION THREAD

**Discussion by joseph1267** [strongswan/strongswan#761] [Q&A]
Hi Strongswan team,
After the tunnel establishment I have
nat1 as 10.0.0.8 --- 192.168.86.38 === 192.168.86.39 --- 10.0.0.10
At this point, the server side (free5gc n3iwf) will send a Create Child SA message with the incrementing MID (6 at this point) to the strongswan to add a additional Child SA with new TSi (60.60.0.1) and TSr (192.168.82.5). But whenever the Create Child SA message comes, Strongswan expects the MID to be 0 and the Create Child SA processing is being dropped since it's MID is 6. I saw the RFC and there were three kind, 
1. Creating New Child SAs with the CREATE_CHILD_SA Exchange
2. Rekeying IKE SAs with the CREATE_CHILD_SA Exchange
3. Rekeying Child SAs with the CREATE_CHILD_SA Exchange

In my case, is it possible to create new child SA with the Create Child SA exchange? or does the Create Child SA exchange message considered Rekeying the Child SAs? 
And in all three cases, does Strongswan have different behaviors?
Thank you

**Comment by tobiasbrunner** [comment]
Please stop reporting bugs in free5gc here.

**Reply by joseph1267** [reply]
Sorry, but this was not a free5gc question, it was about the create child sa... 
When Strongswan receives the Create Child SA payload, will it process the Create Child Sa as Rekeying of Child SA or Creating a New Child SAs, or Rekeying the IKE SAs?

**Reply by tobiasbrunner** [reply]
It does the appropriate thing as per RFC.
